packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts

```diff
@@ -16,7 +16,8 @@ import {
 } from "../styles/recipes";
 import { heightNumber } from "../styles/size";
 
-export const AccordionItemStyles = css`
+export const AccordionItemStyles = (context, definition) =>
```

Comment:
After reviewing the whole PR I think it was a good idea to take this approach consistently across styles and templates, even though it's technically not needed for most cases. This will enable us to avoid a breaking change if we need to switch any styles/templates to take advantage of this, and it will be consistent across the library.

@nicholasrice This does present a bit of a challenge for someone who wants to re-use and then extend the styles. It's still doable since they could write their own function that calls the original to get its styles and the composes them with another set of styles. But I wonder if we could make this nicer. What if we created something like `FoundationStyles` that let you pass it the same function as there is here. It could return a function, but one augmented with some sort of `extends` API to make it easy for folks to extend the styles without having to do all the wrapping, etc. themselves. Just thinking.

Reply:
I wonder if "FAST..." is unnecessary?  The project and context is already known as FAST.  In essence, we could eliminate duplicative naming and just go with "Anchor" in the form of whatever casing we choose to consistently apply.

Reply:
We'd need to go with lower case if we did that otherwise we'd have a function with the same name as the class for a component. Not a bad think to do this in my opinion. It would mean that if you needed to switch design systems, you wouldn't need to change much code. You would just import the registry functions from a different package, and then use the API to change up element prefix. So, that would basically be a two line code change to switch from fast to fluent, for example.

Reply:
> After reviewing the whole PR I think it was a good idea to take this approach consistently across styles and templates, even though it's technically not needed for most cases. This will enable us to avoid a breaking change if we need to switch any styles/templates to take advantage of this, and it will be consistent across the library.

This was my thought as well.

> We'd need to go with lower case if we did that otherwise we'd have a function with the same name as the class for a component. Not a bad thing to do in my opinion. It would mean that if you needed to switch design systems, you wouldn't need to change much code. You would just import the registry functions from a different package, and then use the API to change up element prefix. So, that would basically be a two line code change to switch from fast to fluent, for example.

I'm hesitant if we should keep this line of thinking - or I'm at least hesitant that it should factor significantly into this decision at this time. I'm fine if we re-export without the prefix, but I don't know that we should do it under the guise that it is easier to swap between actual design system implementations. We know how two currently are being built, and we know the ways they are similar - but it doesn't mean that it's going to be a common pattern.

All that said, I'm open to whatever the "right" thing is here, but we should decide quickly if we can :)

Reply:
I don't have a strong opinion on the prefix really. It's loosely held. The only place where the names of these functions matter is in the registration code, which is typically isolated to one file in most applications. So, even if swapping the design system, one could do a simple search and replace to fix the function names if that was the only difference.

I do think we should probably go with lowercase though, since these are functions and not classes.

Reply:
@EisenbergEffect @awentzel I'm moving to lowercase - I started down the road of removing this as I think there's a good argument about it, but we do run into two issues where I think the prefix is helpful:

1. Reserved words in JS. We have a component named `switch` for instance and `option` which are both reserved in terms of either a JS class or method. The prefix helps avoid collision here where a prefix would be required.
2. In certain places we extend component classes, such as "Anchor" or "Button" to add additional logic, etc. In cases like this something like "Anchor" is our new Foundation element for the fast-components set. In this case we would have two exports `Anchor` and `anchor`. The prefix will help create some clarity here.

For the two reasons above, I think for the time being we should keep the prefix.

Reply:
Sounds good to me.

Reply:
@EisenbergEffect yea that was my thought as well; the function type allows extension or replacement. We can do work to make it more straight forward but the core functionality is there.